test/NormalJob.test.ts

```diff
@@ -0,0 +1,45 @@
+import { Job } from '../src/Job'
+import { NormalJob } from '../src/NormalJob'
+
+describe('NormalJob', () => {
+	it('returns the correct name', () => {
+		const job = new NormalJob('jobName')
+		expect(job.getName()).toEqual('jobName')
+	})
+
+	it('returns the correct number of prerequisites', () => {
+		const sourceJob = new NormalJob('', new Set([]))
+		expect(sourceJob.getNumPrerequisites()).toEqual(0)
+
+		const nonsourceJob = new NormalJob(
+			'',
+			new Set([new NormalJob(''), new NormalJob(''), new NormalJob('')]),
+		)
+		expect(nonsourceJob.getNumPrerequisites()).toEqual(3)
+	})
+
+	it("doesn't allow direct member access", () => {
+		const toDelete = new NormalJob('asd')
+		const prerequisites = new Set([toDelete])
+
+		const job = new NormalJob('job', prerequisites)
+		prerequisites.delete(toDelete)
+		expect(job.getNumPrerequisites()).toEqual(1)
+	})
+
+	it('returns the correct prerequisites', () => {
+		const prerequisites = new Set<Job>([
+			new NormalJob('1'),
+			new NormalJob('2'),
+			new NormalJob('3'),
+		])
+
+		const job = new NormalJob('job', prerequisites)
+
+		for (const prerequisite of job.getPrerequisitesIterable()) {
+			prerequisites.delete(prerequisite)
+		}
+
+		expect(prerequisites.size).toEqual(0)
```

Comment:
Simpler test: just create a set from this iterable, then expect it to be equal to `prerequisites`.

That would also work in case NormalJob modifies prerequisites (I know it doesn't, because of the previous test case, but better to be safe).

Reply:
I don't see a way to check `Set` equality that's more efficient than this. Since this is just a test, it could still be better if its more elegant/readable, but unfortunately I haven't found an elegant way to check `Set` equality.

Reply:
I believe `toEqual` in jest does deep equality, so it can handle any JS builtin types (including Set).